src/crimson/osd/pg.cc

```diff
@@ -975,9 +984,9 @@ PG::do_osd_ops_execute(
   FailureFunc&& failure_func)
 {
   assert(ox);
-  auto rollbacker = ox->create_rollbacker([this] (auto& obc) {
-    return obc_loader.reload_obc(obc).handle_error_interruptible(
-      load_obc_ertr::assert_all{"can't live with object state messed up"});
+  auto rollbacker = ox->create_rollbacker(
+    [object_context=duplicate_obc(obc)] (auto& obc) mutable {
+    obc->update_from(*object_context);
```

Comment:
@rzarzynski shared some concerns with me regarding avoiding the reload here. We'll need to verify that the actual obc stored in the registry (`obc_registry`) will be updated accordingly so there won't be any inconsistencies between instances of the same obc.
I'm approving in order to schedule testing, we should probably have another look before merging though.